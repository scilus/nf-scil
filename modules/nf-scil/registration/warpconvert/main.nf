process REGISTRATION_WARPCONVERT {
    tag "$meta.id"
    label 'process_single'

    container "freesurfer/synthmorph:latest"
    containerOptions "--entrypoint ''"

    input:
    tuple val(meta), path(deform), path(affine), path(source) /* optional, value = [] */, path(target) /* optional, value = [] */, path(fs_license) /* optional, value = [] */

    output:
    tuple val(meta), path("*.txt"), emit: init_transform
    tuple val(meta), path("*.nii.gz"), emit: deform_transform
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    //For arguments definition, lta_convert -h
    def invert = task.ext.invert ? "--invert" : ""
    def source_geometry_init = "$source" ? "--src" + "$source" : ""
    def target_geometry_init = "$target" ? "--trg" + "$target" : ""
    def out_format_init = task.ext.out_format_init ? "--out" + task.ext.out_format_init : "--outitk"

    //For arguments definition, mri_warp_convert -h
    def source_geometry_deform = "$source" ? "--insrcgeom" + "$source" : ""
    def in_format_deform = task.ext.in_format_deform ? "--in" + task.ext.in_format_deform : "--inras"
    def out_format_deform = task.ext.out_format_deform ? "--out" + task.ext.out_format_deform : "--outitk"
    def downsample = task.ext.downsample ? "--downsample" : ""


    """
    lta_convert $source_geometry_init $target_geometry_init $out_format_init ${prefix}__init_warp.txt
    mri_warp_convert $source_geometry_deform $downsample $out_format_deform  ${prefix}__deform_warp.nii.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        Freesurfer: 7.4.1
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    lta_convert -h
    mri_warp_convert -h

    touch ${prefix}__init_transform.txt
    touch ${precix}__deform_transform.nii.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        Freesurfer: 7.4.1
    END_VERSIONS
    """
}
