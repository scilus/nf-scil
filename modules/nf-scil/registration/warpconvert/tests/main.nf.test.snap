{
    "registration - warpconvert - fsants": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__init_warp.txt:md5,77b7540118a282f40c653b11eeeb5bd1"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__deform_warp.nii.gz:md5,798ae9c709eb58bf0eb5013f23bcb913"
                    ]
                ],
                "2": [
                    "versions.yml:md5,2960624ce93ca37d3969a87aad502cfc"
                ],
                "deform_transform": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__deform_warp.nii.gz:md5,798ae9c709eb58bf0eb5013f23bcb913"
                    ]
                ],
                "init_transform": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__init_warp.txt:md5,77b7540118a282f40c653b11eeeb5bd1"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2960624ce93ca37d3969a87aad502cfc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-06-07T12:32:53.798876"
    },
    "registration - warpconvert - default": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__init_warp.txt:md5,77b7540118a282f40c653b11eeeb5bd1"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__deform_warp.nii.gz:md5,798ae9c709eb58bf0eb5013f23bcb913"
                    ]
                ],
                "2": [
                    "versions.yml:md5,2960624ce93ca37d3969a87aad502cfc"
                ],
                "deform_transform": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__deform_warp.nii.gz:md5,798ae9c709eb58bf0eb5013f23bcb913"
                    ]
                ],
                "init_transform": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__init_warp.txt:md5,77b7540118a282f40c653b11eeeb5bd1"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2960624ce93ca37d3969a87aad502cfc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-06-07T12:32:33.937568"
    }
}